packages/components/src/components/domainproperties/models.ts

```diff
@@ -1405,3 +1406,36 @@ export interface IAppDomainHeader {
 }
 
 export type DomainPanelStatus = 'INPROGRESS' | 'TODO' | 'COMPLETE' | 'NONE';
+
+/**
+ * General object to describe a DomainKind
+ *
+ * @property domainDesign The fields found on all items of this domain (e.g., copy per item)
+ * @property options The fields and properties shared by this type (e.g., one copy) ('options' used to match existing LKS api)
+ * @property domainKindName The name of the domainkind type this represents, currently supported can be found in Domain.KINDS
+ */
+export class DomainDetails {
+    domainDesign: DomainDesign;
+    options: Map<string, any>;
+    domainKindName: string;
+
+    static create(rawDesign: Map<string, any> = Map(), domainKindType: string = Domain.KINDS.UNKNOWN): DomainDetails {
```

Comment:
Was this class meant to extend record? Right now this create method creates plain JS objects, it does not actually instantiate DomainDetails classes. There is no constructor, so you can't instantiate this class.

Reply:
extended.